### ISSUE TITLE
Publish at least latest stable branch of doxygen docs

### ISSUE THREAD

**Issue by robe2** [libgeos/geos#828]
One of my clients mentioned this.

The website only publishes the not-yet released version of GEOS developer docs.
So they got all excited seeing GEOSLineSubstring_r  in the [docs](https://libgeos.org/doxygen/geos__c_8h.html)  only to be disappointed that it's not in a released version.

I think we should list at the very least, the latest stable version of the docs on the website.

I think this might be slightly tricky though since our website lives in the develop branch.

I'll investigate and try to implement.

**Comment by mwtoews** [comment]
The CGAL folks have figured out multi-version doxygen docs [here](https://doc.cgal.org/latest/Manual/index.html) (top left dropdown for CGAL Version). See how they did it. Maybe somewhere [here](https://github.com/CGAL/cgal-web)?

**Comment by robe2** [comment]
I couldn't quite follow what they were doing, but I think ours will be easier.

My guess is I can just define a .github/workflow/ similar to the web one here - https://github.com/libgeos/geos/blob/main/.github/workflows/web.yml

But will call build_doxygen.yml and will only have the lineitems for building doxygen and always publish to the website, but in different folder for each branch.

**Comment by robe2** [comment]
I hardcoded the 3.11 which is slightly annoying but I was too tired to figure out how to variablize it based on branch.  Then again maybe it should be hard-coded to prevent test branches from deploying.  I'm satisfied with this for now.

I changed the page to have both dev and 3.11 branch -  https://libgeos.org/usage/doxygen/

Might be nice to have a 3.10 too since that seems to be more in the wild than 3.11.
